tuning/acceleration/plugins/framework_plugin_unsloth_autogptq.py

```diff
@@ -0,0 +1,150 @@
+
+def parse_configuration():
+    pass
+
+import torch
+from transformers import TrainingArguments
+from peft import LoraConfig
+from peft.tuners.lora.model import LoraModel
+from auto_gptq.utils.peft_utils import GPTQLoraModel
+from unsloth import FastLanguageModel
+from unsloth.utils.modeling import QuantizationMethod
+from unsloth.gptq.triton.layers import GPTQuantLinear
+
+from types import MethodType
+from typing import Tuple, List, Dict
+
+from .framework_plugin import AccelerationPlugin
+from .framework_plugin_autogptq import _replace_module, LoraLinearGPTQ
+
+# def _dispatch_lora_linear_for_gptq(
+def _create_new_module_triton(
+    lora_config: LoraConfig,
+    adapter_name: str,
+    target: torch.nn.Module,
+    **kwargs,
+):
+    # if the base layer module matches a supported class, dispatch the lora linear
+    # to be installed
+    new_module = target
+    if (
+        isinstance(target, GPTQuantLinear)
+    ):
+        new_module = LoraLinearGPTQ(target, adapter_name, **kwargs)
+    # if module cannot be found, return None which results in a raise in the call-stack
+    return new_module
+
+class UnslothAutoGPTQAccelerationPlugin(AccelerationPlugin):
+    
+    require_packages = ['auto_gptq', 'unsloth']
+    restricted_model_archs = ['MixtralForCausalLM', 'LlamaForCausalLM', 'MistralForCausalLM', 'GemmaForCausalLM']
+
+    def __init__(self, configurations: Dict[str, Dict]):
+        super().__init__(configurations)
+
+        # just do checking, nothing must to configure at this point
+        # if need to configure then do something like this:
+        # self.kernel = self._get_config_value("peft.kernel")
+        self._check_config_equal(key="peft.quantization", value="unsloth")
+        self._check_config_equal(key="peft.quantization.unsloth.base_layer", value="auto_gptq")
+        self._check_config_equal(key="peft.quantization.unsloth.kernel", value="triton_v2")
+
+    def model_loader(self, model_name: str, **kwargs):
+        # 1. Load the gptq base model through unsloth FastLanguageModel
+        model, _ = FastLanguageModel.from_pretrained(
+            model_name,
+            quantization_method=QuantizationMethod.GPTQ,
+            device_map="auto",
+        )
+
+        # 2. Depending on the model, replace appropriate CUDA kernel with TritonV2 kernel 
```

Comment:
```suggestion
        # 2. Depending on the model, replace appropriate CUDA kernel with TritonV2 kernel 
        # - `FastLanguageModel` above will load a "default" cuda linear kernel. 
        # - the idea is to load the appropriate "default" kernel (for each model), then follow up with an injection
        #.  via the `inject_to_model(..., target_module_type=DefaultKernel)` method.
```

Reply:
improved the comments for now. This logic is not very satisfying, but ok for now.